src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -531,8 +547,38 @@ CreateShardsOnWorkers(Oid distributedRelationId, List *shardPlacements,
 			shardIndex = ShardIndex(shardInterval);
 		}
 
-		connection = GetPlacementConnection(connectionFlags, shardPlacement,
-											placementOwner);
+		/*
+		 * For partitions, make sure that we mark the parent table relation access
+		 * with DDL. This is only important for parallel relation access in transaction
+		 * blocks, thus check useExclusiveConnection and transaction block as well.
+		 */
+		if ((ShouldRecordRelationAccess() && useExclusiveConnection) &&
+			alterTableAttachPartitionCommand != NULL)
```

Comment:
same here (alterTableAttachPartitionCommand != NULL) meaning this table is a partition of a distributed partitioned table.